src/thread_win32.h

```diff
@@ -67,4 +67,47 @@ typedef std::condition_variable ConditionVariable;
 
 #endif
 
+#if !defined(_MSC_VER) // defined(__APPLE__) FIXME when done with testing
+
+#include <pthread.h>
+
+/// On OSX threads other than the main thread are created with a reduced stack
+/// size of 512KB by default, this is dangerously low for deep searches, so
+/// adjust it to TH_STACK_SIZE. The implementation calls pthread_create() with
+/// proper stack size parameter.
+
+static const size_t TH_STACK_SIZE = 2 * 1024 * 1024;
+
+template <class Fun, class Obj, class P = std::pair<Fun, Obj*>>
+void* start_routine(void* ptr)
+{
+   P* p = reinterpret_cast<P*>(ptr);
+   (p->second->*(p->first))(); // Call member function pointer
+   delete p;
+   return NULL;
+}
+
+class NativeThread {
+
+   pthread_t thread;
+
+public:
+  template<class Fun, class Obj>
+  explicit NativeThread(Fun fun, Obj* obj) {
+    auto ptr = new std::pair<Fun, Obj*>(fun, obj); // Should survive to us
+    pthread_attr_t attr_storage, *attr = &attr_storage;
+    pthread_attr_init(attr);
+    pthread_attr_setstacksize(attr, TH_STACK_SIZE);
+    pthread_create(&thread, attr, start_routine<Fun, Obj>, (void*)ptr);
```

Comment:
should there be a check of the exit code, in case such a thread can not be created (i.e. what happens if the user limited the stacksize).

Reply:
It crashes, as it is the case for std::thread too, where there is nothing upon creation